src/modules/tools/laser/Laser.cpp

```diff
@@ -224,7 +238,19 @@ bool Laser::get_laser_power(float& power) const
 // called every millisecond from timer ISR
 uint32_t Laser::set_proportional_power(uint32_t dummy)
 {
-    if(manual_fire) return 0;
```

Comment:
This is not called every millisecond, the comment is incorrect.

Reply:
Oh... 
Btw another comment is a bit confusing as well, please correct me if i am wrong
    // no point in updating the power more than the PWM frequency, but no more than 1KHz
while it means 
    // no point in updating the power more than the PWM frequency, but not faster than 1KHz
As for example if we have fast laser and we set it 20us, it will pick minimum between 1000 and 50000, which is 1khz(firing 1ms), if we set 400us (as my particular case), it is still within range.

But yes, if someone will set >1000us it will be a different story.
Probably then i just round it to nearest value, so if we assume that by some reason user set 100hz, then we recalculate and send him in warning that it will be minimal period - 10ms, and if he set  21ms - 20ms. (I guess not flooring, but choosing less value is safer).
Is it fine like this?

Reply:
Ca n you check please if it is fine now?